hazelcast/src/test/java/com/hazelcast/sql/impl/expression/datetime/FormatterTest.java

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.datetime;
+
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.RoundingMode;
+import java.text.DecimalFormat;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.Year;
+import java.time.YearMonth;
+import java.time.ZoneOffset;
+import java.util.Locale;
+
+import static com.hazelcast.sql.impl.expression.datetime.Formatter.forDates;
+import static com.hazelcast.sql.impl.expression.datetime.Formatter.forNumbers;
+import static java.lang.Double.NEGATIVE_INFINITY;
+import static java.lang.Double.NaN;
+import static java.util.Locale.US;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+@SuppressWarnings("checkstyle:ParenPad")
+public class FormatterTest {
+    private static final Locale TR = new Locale("tr", "TR");
+    private static int jdk;
+
+    @BeforeClass
+    public static void setup() {
+        String[] version = System.getProperty("java.version").split("\\.");
+        jdk = Integer.parseInt(version[version[0].equals("1") ? 1 : 0]);
+    }
+
+    @Test
+    public void testDates() {
+        Formatter f = forDates("FMDay, Mon DDth, YYYY");
```

Comment:
We should also test empty and null patterns as edge cases and match PG behavior in those cases.